nexus/src/app/session.rs

```diff
@@ -34,40 +35,13 @@ fn generate_session_token() -> String {
 }
 
 impl super::Nexus {
-    async fn login_allowed(
-        &self,
-        opctx: &OpContext,
-        silo_user_id: Uuid,
-    ) -> Result<bool, Error> {
-        // Was this silo user deleted?
-        let fetch_result = LookupPath::new(opctx, &self.db_datastore)
-            .silo_user_id(silo_user_id)
-            .fetch()
-            .await;
-
-        match fetch_result {
-            // if the silo user was deleted, they're not allowed to log in :)
-            Err(Error::ObjectNotFound { .. }) => Ok(false),
-            Err(e) => Err(e),
-            // they're allowed
-            Ok(_) => Ok(true),
-        }
-    }
-
```

Comment:
looking up the user by ID when either have the DB model on hand or we don't at the call side makes no sense unless we're trying to rule out the user being deleted in the 1ms between the retrieval and now

Reply:
I traced this waaaay back, and when this was added, there was _only_ spoof login and the `user_id` was hard-coded. I am confident it is no longer applicable. The methods used to authenticate users for local and SAML login both do the user lookup themselves.

https://github.com/oxidecomputer/omicron/blob/9d548050c647d301dee91da4802de629ee22b122/nexus/src/external_api/console_api.rs#L48-L70